src/LottoConsoleGame.js

```diff
@@ -0,0 +1,123 @@
+import { LOTTO, RANKING_THRESHOLD, GAME_COMMAND } from './constants';
+import lottoGameCalculator from './domain/lottoGameCalculator';
+import lottoGameValidator from './domain/lottoGameValidator';
+import Lotto from './domain/models/Lotto';
+import generateRandomNumber from './utils/generateRandomNumber';
+import Interface from './view/Interface';
+import outputView from './view/outputView';
+
+class LottoConsoleGame {
+  #io;
+  #lottos;
+
+  constructor() {
+    this.#io = new Interface();
+    this.#lottos = [];
+  }
+
+  async play() {
+    const pruchaseAmount = await this.readPurchaseAmount();
+    this.buyLottos(pruchaseAmount);
+    this.printLottos();
+
+    const winningNumbers = await this.readWinningNumbers();
+    const bonusNumber = await this.readBonusNumber(winningNumbers);
+    this.printStatistics(pruchaseAmount, this.makeRankings(winningNumbers, bonusNumber));
+
+    this.decideReplay(await this.readGameCommand());
+  }
+
+  buyLottos(purchaseAmount) {
+    while (this.#lottos.length < purchaseAmount / LOTTO.price) {
+      this.#lottos.push(this.buyLotto());
+    }
+  }
+
+  buyLotto() {
+    const randomNumbers = [];
+    while (randomNumbers.length < LOTTO.numbersLength) {
+      const randomNumber = generateRandomNumber(LOTTO.minNumber, LOTTO.maxNumber);
+      if (!randomNumbers.includes(randomNumber)) randomNumbers.push(randomNumber);
+    }
+
+    return new Lotto(randomNumbers.sort((a, b) => a - b));
+  }
+
+  makeRankings(winningNumbers, bonusNumber) {
+    const rankings = [];
+    this.#lottos.forEach((lotto) => {
+      const matchCount = lotto.calculateMatchCount(winningNumbers);
+      if (matchCount >= RANKING_THRESHOLD) {
+        rankings.push(lotto.calculateRanking(matchCount, bonusNumber));
+      }
+    });
```

Comment:
Array.filter 함수를 사용하면 더 간결한 형태로 바꿀 수 있겠어요

Reply:
사실 이 부분의 depth가 2이라서 개선이 필요했는데 감사합니다!
아래처럼 바꿀 수 있을 것 같습니다.
다만 Lotto.calculateMatchCount 가 2번 호출되는데 이 부분은 어쩔 수 없는 것 같습니다.
```suggestion
    const rankings = this.#lottos
      .filter((lotto) => lotto.calculateMatchCount(winningNumbers) >= RANKING_THRESHOLD)
      .map((lotto) => lotto.calculateRanking(lotto.calculateMatchCount(winningNumbers), bonusNumber));
```